src/mono/mono/utils/mono-proclib.c

```diff
@@ -190,3 +191,44 @@ mono_cpu_count (void)
 	return 1;
 #endif
 }
+
+/**
+ * mono_cpu_limit:
+ * \returns the number of processors available to this process
+ */
+int
+mono_cpu_limit (void)
+{
+#if defined(HOST_WIN32)
+       return mono_cpu_count();
+#else
+       static int limit = -1;  /* Value will be cached for future calls */
+
+       /*
+        * If 1st time through then check if user has mandated a value and use it,
+        * otherwise we check for any cgroup limit and use the min of actual number
+        * and that limit
+        */
+       if (limit == -1) {
+               char *dotnetProcCnt = getenv("DOTNET_PROCESSOR_COUNT");
+               if (dotnetProcCnt != NULL) {
+                       errno = 0;
+                       limit = strtol(dotnetProcCnt, NULL, 0);
```

Comment:
implicit cast `long` to `int`. is that ok?

Reply:
I would use `atoi()` but there's no error checking. Do you want an explicit cast?

Reply:
well that's my question: should we just cast? do we need to check that the value's in range?